backend/battle/battles/base_stats.py

```diff
@@ -33,3 +34,18 @@ def sum_all_pokemons(pokemons):
 def sumValid(pokemon):
     sum_result = pokemon["attack"] + pokemon["defense"] + pokemon["hp"]
     return sum_result
+
+
+def verify_team_exists(battle, trainer):
+    pokemons_team = {
+        "pokemon_1": 0,
+        "pokemon_2": 0,
+        "pokemon_3": 0,
+    }
+    team = Team.objects.filter(battle=battle, trainer=trainer)
```

Comment:
Use `Team.objects.prefetch_related('pokemons').filter(battle=battle, trainer=trainer)` to avoid an additional query for each Pokemon.

Reply:
Thanks, the prefetch_related and select_related' approach is still a bit confusing to me

Reply:
No worries, check http://pythonclub.com.br/django-introducao-queries.html

Reply:
thanks :)